cmd/cli/app/repo/repo_register.go

```diff
@@ -67,6 +66,23 @@ func RegisterCmd(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn)
 		return cli.MessageAndError("Error getting list of repos", err)
 	}
 
+	alreadyRegisteredReposSet := sets.New[string]()
+	for _, repo := range alreadyRegisteredRepos.Results {
+		alreadyRegisteredReposSet.Insert(cli.GetRepositoryName(repo.Owner, repo.Name))
+	}
+
+	inputRepositories := strings.Split(repoList, ",")
+	unregisteredInputRepos, warnings := getUnregisteredInputRepos(inputRepositories, alreadyRegisteredReposSet)
+
+	for _, warning := range warnings {
+		cmd.Println(warning)
+	}
+
+	allInputReposRegistered := inputRepositories[0] != "" && len(unregisteredInputRepos) == 0
+	if allInputReposRegistered {
+		return nil
+	}
+
 	// Get a list of remote repos
 	remoteListResp, err := client.ListRemoteRepositoriesFromProvider(ctx, &minderv1.ListRemoteRepositoriesFromProviderRequest{
 		Context: &minderv1.Context{Provider: &provider, Project: &project},
```

Comment:
Reading all this again, it feels like this would be better to split behavior based on `repoList`:

```golang
unregisteredRepos []string
if repoList != "" {
  // split repoList, remove already-registered repos
  // add remainder to unregisteredRepos
  if len(unregisteredRepos) == 0 {
    cmd.Println("All repos already registered, nothing to be done")
    return nil
  }
} else {
  // call ListRemoteRepositoriesFromProvider, select from menu, etc
}

reposToAdd []*minderv1.UpstreamRepositoryRef
for _, selectedRepo := range unregisteredRepos {
  // Parse repo name, store in reposToAdd
}
```

Reply:
BTW - I think the factoring and testing you're doing here is great, and I'm okay leaving this PR as-is for the testing improvement.  But I'm also looking at the overall code flow, and it does feel like there's a bunch of weird edge cases here that it would be great if you wanted to clean up.

Reply:
> Reading all this again, it feels like this would be better to split behavior based on repoList:

I modified it to use `repoList`. Indeed, accessing array indices for boolean conditions is not the best solution. I have simplified the program flow by splitting it into functions. 

> bunch of weird edge cases here that it would be great if you wanted to clean up.

I refactored the code for visual simplification. I feel all cases are covered except the survey prompt, invoking RPCs, and a few `len` checks. Please let me know if you'd like me to add some mocking or any other test.